### PR TITLE
CBG-1646 Add a flag to non-active leaf revisions indicating whether they include v2 attachments

### PR DESCRIPTION
**Pull request by sarathkumarsivan**
Changest to include a flag to non-active leaf revisions indicating whether they include v2 attachments.

### CODE REVIEW

File: db/revtree.go [hunk 1]
```diff
@@ -22,13 +22,14 @@ type RevKey string
...
-	Body     []byte // Used when revision body stored inline (stores bodies)
-	Channels base.Set
+	ID           string
+	Parent       string
+	BodyKey      string // Used when revision body stored externally (doc key used for external storage)
+	Deleted      bool
+	depth        uint32
+	Body         []byte // Used when revision body stored inline (stores bodies)
+	Channels     base.Set
+	IncludeV2Att bool
```
Comment (adamcfraser) [inline db/revtree.go]: Change to HasAttachments throughout, to indicate that this is a status, not an action.

File: db/revtree.go [hunk 2]
```diff
@@ -42,13 +43,14 @@ type RevTree map[string]*RevInfo
...
-	BodyKeyMap map[string]string `json:"bodyKeyMap,omitempty"` // Keys of revision bodies stored in external documents
-	Channels   []base.Set        `json:"channels"`
+	Revs        []string          `json:"revs"`                 // The revision IDs
+	Parents     []int             `json:"parents"`              // Index of parent of each revision (-1 if root)
+	Deleted     []int             `json:"deleted,omitempty"`    // Indexes of revisions that are deletions
+	Bodies_Old  []string          `json:"bodies,omitempty"`     // JSON of each revision (legacy)
+	BodyMap     map[string]string `json:"bodymap,omitempty"`    // JSON of each revision
+	BodyKeyMap  map[string]string `json:"bodyKeyMap,omitempty"` // Keys of revision bodies stored in external documents
+	Channels    []base.Set        `json:"channels"`
+	Attachments []int             `json:"attachments,omitempty"` // Indexes of revisions that has attachments
```
Comment (adamcfraser) [inline db/revtree.go]: Again, HasAttachments.

File: db/crud_test.go [hunk 3]
```diff
@@ -102,6 +102,71 @@ func TestRevisionCacheLoad(t *testing.T) {
...
+		"digest": "sha1-Kq5sNclPz7QV2+lfQIuc6R7oRu0=", "length": 11, "revpos": 2, "stub": true, "ver": 2}}
+	assert.Equal(t, rev2b_body, gotbody)
+
+	// Retrieve the raw document, and verify 2-a isn't stored inline
+	log.Printf("Retrieve doc, verify rev 2-a not inline")
+	revTree, err := getRevTreeList(db.Bucket, "doc1", db.UseXattrs())
+	assert.NoError(t, err, "Couldn't get revtree for raw document")
+	assert.Equal(t, 0, len(revTree.BodyMap))
+	assert.Equal(t, 1, len(revTree.BodyKeyMap))
+	assert.Equal(t, 1, len(revTree.Attachments))
```
Comment (adamcfraser) [inline db/crud_test.go]: Can you add a test case for legacy attachments, to ensure they aren't being flagged when storing the old revision body.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/crud_test.go (hunk 3 above)
Comment (sarathkumarsivan) [reply db/crud_test.go]: Sure, an additional unit test has been included for legacy attachments.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**